engine/lib/phx/src/engine/engine.rs

```diff
@@ -483,4 +521,276 @@ impl Engine {
+
+    /// Stop the multithreaded render system.
+    ///
+    /// This shuts down the render thread and returns the GL context
+    /// to the main thread for direct rendering.
+    #[bind(name = "StopRenderThread")]
+    pub fn stop_render_thread_ffi(&mut self) {
+        self.stop_render_thread()
+    }
+
+    /// Check if the render thread is currently active.
+    #[bind(name = "IsRenderThreadActive")]
+    pub fn is_render_thread_active(&self) -> bool {
+        self.render_context.is_active()
+    }
+
+    /// Get total commands processed by the render thread.
+    #[bind(name = "GetRenderThreadCommands")]
+    pub fn get_render_thread_commands(&self) -> u64 {
+        self.render_context.get_commands_processed()
+    }
+
+    /// Get total draw calls executed by the render thread.
+    #[bind(name = "GetRenderThreadDrawCalls")]
+    pub fn get_render_thread_draw_calls(&self) -> u64 {
+        self.render_context.get_draw_calls()
+    }
+
+    /// Get total state changes on the render thread.
+    #[bind(name = "GetRenderThreadStateChanges")]
+    pub fn get_render_thread_state_changes(&self) -> u64 {
+        self.render_context.get_state_changes()
+    }
+
+    /// Get total frames rendered by the render thread.
+    #[bind(name = "GetRenderThreadFrameCount")]
+    pub fn get_render_thread_frame_count(&self) -> u64 {
+        self.render_context.get_frame_count()
+    }
+
+    /// Get the last frame render time in milliseconds.
+    #[bind(name = "GetRenderThreadFrameTimeMs")]
+    pub fn get_render_thread_frame_time_ms(&self) -> f64 {
+        self.render_context.get_last_frame_time_ms()
+    }
+
+    /// Get commands processed in the last frame.
+    #[bind(name = "GetRenderThreadCommandsPerFrame")]
+    pub fn get_render_thread_commands_per_frame(&self) -> u64 {
+        self.render_context.get_commands_last_frame()
+    }
+
+    /// Get draw calls executed in the last frame.
+    #[bind(name = "GetRenderThreadDrawCallsPerFrame")]
+    pub fn get_render_thread_draw_calls_per_frame(&self) -> u64 {
+        self.render_context.get_draw_calls_last_frame()
+    }
+
+    /// Get total texture binds skipped due to caching (cumulative).
+    #[bind(name = "GetRenderThreadTextureBindsSkipped")]
+    pub fn get_render_thread_texture_binds_skipped(&self) -> u64 {
+        self.render_context.get_texture_binds_skipped()
+    }
+
+    /// Get main thread wait time in milliseconds (time spent waiting for render thread).
+    #[bind(name = "GetMainThreadWaitTimeMs")]
+    pub fn get_main_thread_wait_time_ms(&self) -> f64 {
+        self.render_context.get_main_thread_wait_ms()
+    }
+
+    /// Get current frames in flight (submitted but not yet rendered).
+    #[bind(name = "GetFramesInFlight")]
+    pub fn get_frames_in_flight(&self) -> u64 {
+        self.render_context.get_frames_in_flight()
+    }
+
+    /// Get the number of CPU cores available for worker threads.
+    #[bind(name = "GetCpuCount")]
+    pub fn get_cpu_count(&self) -> u32 {
```

Comment:
This and following methods are not used anywhere.
If it's planned for the future then maybe better to create a dedicated struct for it, i.e. System/SystemInfo?